apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -159,6 +159,11 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
   const hasAppendedInitialMessage = useRef<boolean>(false);
 
   const lessonPlanSnapshot = useRef<LooseLessonPlan>({});
+  const { lessonPlanManager, lessonPlan } = useLessonPlanManager();
+
+  const [currentIteration, setCurrentIteration] = useState<number | undefined>(
```

Comment:
Each time we save the chat, we increment this number. We can then use it to compare the current lesson plan with what has come down the wire and only update it if it has incremented.